tests/typechecker/function_base_class_set_argument_derived_given.jakt

```diff
@@ -0,0 +1,24 @@
+/// Expect:
+/// - output: "PASS\n"
+
+trait Hashable<Output> {
+    fn hash(self) -> Output
+}
+
```

Comment:
uhh...this definitely shouldn't be accepted, this is not the "correct" definition of Hashable as assumed by the compiler

Reply:
I mean, defining that trait is fine, but `{T}` where `T implements(Hashable<i64>)` should not be accepted

Reply:
Hm, so how this should look like?

Reply:
`Hashable` is already in scope, imported from the prelude, it should just `implements(Hashable)`

Reply:
It took me a sec to figure out I'm missing `this` in `public fn hash(this) => 42u32`
Now I know why it didn't work for me before :p